Sources/SwiftDocC/Infrastructure/Topic Graph/TopicGraph.swift

```diff
@@ -70,7 +70,7 @@ struct TopicGraph {
         }
         
         /// The reference to the `DocumentationNode` this node represents.
-        let reference: ResolvedTopicReference
+        var reference: ResolvedTopicReference
```

Comment:
It seems strange to me to make only the reference mutable. Should we just be making a new DocumentationNode when we need to modify the reference? That seems like a core part of defines the node.

Reply:
I think since `Node` is a reference type, it makes sense to avoid accidental mutation. I added an API to create a new node given a new reference, and an API in `TopicGraph` to update a node's reference.